dumb-init.c

```diff
@@ -153,9 +129,22 @@ int main(int argc, char *argv[]) {
 
         execvp(argv[1], &argv[1]);
     } else {
+        pid_t killed_pid;
+        int exit_status, status;
+
         DEBUG("Child spawned with PID %d.\n", child_pid);
-        for (;;) {
-            pause();
+
+        while ((killed_pid = waitpid(-1, &status, 0))) {
```

Comment:
I'm not sure this is an issue, since we no longer actually have a SIGCHLD handler. Instead, we just call `waitpid(-1)` in a loop in `main`.